satpy/tests/reader_tests/conftest.py

```diff
@@ -15,4 +15,243 @@
+# W_XX-EUMETSAT-Darmstadt,SAT,AWS1-MWR-1B-RAD_C_EUMT_20241121085911_G_D_20241109234502_20241110004559_T_N____.nc
+eumetsat_file_pattern = "W_XX-EUMETSAT-Darmstadt,SAT,{platform_name}-MWR-1B-RAD_C_OHB_{processing_time:%Y%m%d%H%M%S}_G_D_{start_time:%Y%m%d%H%M%S}_{end_time:%Y%m%d%H%M%S}_T_B____.nc"  # noqa
+
+esa_file_pattern = "W_XX-OHB-Stockholm,SAT,{platform_name}-MWR-1B-RAD_C_OHB_{processing_time:%Y%m%d%H%M%S}_G_D_{start_time:%Y%m%d%H%M%S}_{end_time:%Y%m%d%H%M%S}_T_B____.nc"  # noqa
+
+esa_l1c_file_pattern = "W_XX-OHB-Stockholm,SAT,{platform_name}-MWR-1C-RAD_C_OHB__{processing_time:%Y%m%d%H%M%S}_G_D_{start_time:%Y%m%d%H%M%S}_{end_time:%Y%m%d%H%M%S}_T_B____.nc"  # noqa
+
+
+rng = np.random.default_rng()
+
+def random_date(start, end):
+    """Create a random datetime between two datetimes."""
+    delta = end - start
+    int_delta = (delta.days * 24 * 60 * 60) + delta.seconds
+    random_second = randrange(int_delta)
+    return start + timedelta(seconds=random_second)
+
+
+@pytest.fixture(scope="session")
+def fake_mwr_data_array():
+    """Return a fake AWS/EPS-Sterna MWR l1b data array."""
+    fake_data_np = rng.integers(0, 700000, size=10*145*19).reshape((10, 145, 19))
+    fake_data_np[0, 0, 0] = -2147483648
+    fake_data_np[1, 0, 0] = 700000 + 10
+    fake_data_np[2, 0, 0] = -10
+    array_dims = ["n_scans", "n_fovs", "n_channels"]
+    return xr.DataArray(fake_data_np, dims=array_dims)
+
+
+def make_fake_angles(geo_size, geo_dims, shape=(10, 145, 4)):
+    """Return fake sun-satellite angle array."""
+    maxval = 36000
+    dummy_array = (np.arange(0, geo_size) * maxval/geo_size).astype("int32")
+    return xr.DataArray(dummy_array.reshape(shape), dims=geo_dims)
+
+
+def make_fake_mwr_lonlats(geo_size, geo_dims):
+    """Return fake geolocation data arrays for all 4 MWR horns."""
+    maxval = 3600000
+    dummy_array = (np.arange(0, geo_size) * maxval/geo_size).astype("int32")
+    fake_lon_data = xr.DataArray(dummy_array.reshape((10, 145, 4)), dims=geo_dims)
+    maxval = 1800000
+    dummy_array = (np.arange(0, geo_size) * maxval/geo_size - maxval/2).astype("int32")
+    fake_lat_data = xr.DataArray(dummy_array.reshape((10, 145, 4)), dims=geo_dims)
+    return (fake_lon_data, fake_lat_data)
+
+
+def make_fake_mwr_l1c_lonlats(geo_size, geo_dims):
+    """Return fake level-1c geolocation data arrays."""
+    maxval = 3600000
+    dummy_array = (np.arange(0, geo_size) * maxval/geo_size).astype("int32")
+    fake_lon_data = xr.DataArray(dummy_array.reshape((10, 145)), dims=geo_dims)
+    maxval = 1800000
+    dummy_array = (np.arange(0, geo_size) * maxval/geo_size - maxval/2).astype("int32")
+    fake_lat_data = xr.DataArray(dummy_array.reshape((10, 145)), dims=geo_dims)
+    return (fake_lon_data, fake_lat_data)
+
+
+@pytest.fixture(scope="module")
+def eps_sterna_mwr_file(tmp_path_factory, fake_mwr_data_array):
+    """Create an EPS-Sterna MWR l1b file."""
+    geo_dims = ["n_scans", "n_fovs", "n_feedhorns"]
+    geo_size = 10*145*4
```

Comment:
Has the pre-commit stuff been run? This should've been caught by the linter.
```suggestion
    geo_size = 10 * 145 * 4
```

Reply:
Yes, I have pre-commit installed and activated!